assets/js/base/components/checkout/form-step/index.js

```diff
@@ -29,9 +29,12 @@ const StepNumber = ( { stepNumber } ) => {
 	);
 };
 
-const StepHeading = ( { title, stepHeadingContent } ) => (
+const StepHeading = ( { title, legend, stepHeadingContent } ) => (
 	<div className="wc-components-checkout-step__heading">
-		<h4 className="wc-components-checkout-step__title">{ title }</h4>
+		<h4 className="wc-components-checkout-step__title">
+			<legend className="screen-reader-text">{ legend || title }</legend>
```

Comment:
Nice!  Like the legend option with the title fallback.

I think this legend cannot be inside an `h4` and needs to come immediately after the opening `<fieldset>` tag to be semantically correct.  Checked using Voiceover and the legend currently isn't read.